framework/bootstrap/http.go

```diff
@@ -15,23 +15,63 @@ import (
 
 // PathHandler for module using
 type PathHandler struct {
-	mux     *http.ServeMux
-	mapping map[string]http.Handler
+	mux           *http.ServeMux
+	mapping       map[string]http.Handler
+	pathRedirects map[string]string
 	sync.RWMutex
 }
 
-func NewPathHandler() *PathHandler {
+func NewPathHandler(pathRedirects map[string]string) *PathHandler {
+	var reversePathRedirects map[string]string
+	if len(pathRedirects) > 0 {
+		reversePathRedirects = make(map[string]string, len(pathRedirects))
+		for k, v := range pathRedirects {
+			if k != v {
+				reversePathRedirects[v] = k
+			}
+		}
+	}
+
 	return &PathHandler{
-		mux:     http.NewServeMux(),
-		mapping: make(map[string]http.Handler),
+		mux:           http.NewServeMux(),
+		mapping:       make(map[string]http.Handler),
+		pathRedirects: reversePathRedirects,
 	}
 }
 
 func (ph *PathHandler) Handle(path string, handler http.Handler) {
-	ph.mux.Handle(path, handler)
+	if path == "" {
+		log.Warn("ignore empty path")
+		return
+	}
+
 	ph.Lock()
+	if _, ok := ph.mapping[path]; ok {
+		ph.Unlock()
+		log.Warnf("ignore dup path %s", path)
+		return
+	}
 	ph.mapping[path] = handler
+
+	var (
+		redir      = ph.pathRedirects[path]
+		redirInUse = redir
+	)
+	if _, ok := ph.mapping[redir]; ok {
+		redirInUse = ""
+	} else {
+		ph.mapping[redir] = handler
+	}
 	ph.Unlock()
+
+	log.Infof("register path %s", path)
+	ph.mux.Handle(path, handler)
+	if redirInUse != "" {
+		log.Infof("register redir path %s", redirInUse)
+		ph.mux.Handle(redir, handler)
+	} else if redir != "" {
+		log.Warnf("ignore dup redir path %s", redir)
```

Comment:
so don't support a->b and c->b ?

Reply:
not really. just in case of both `a->b` and `a->c`